src/core/SideNavigation/SidenavSubmenu.tsx

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/side-navigation.css';
+
+export type SidenavSubmenuProps = {
+  /**
+   * Content of the submenu.
+   */
+  children: React.ReactNode;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Subcomponent to be used in the `submenu` prop of `SideNavigation`.
+ * @example
+ * <SidenavSubmenu>
+ *   <SidenavSubmenuHeader>Documents</SidenavSubmenuHeader>
+ *   <div> ... </div>
+ * </SidenavSubmenu>
+ */
+export const SidenavSubmenu = React.forwardRef<
```

Comment:
I am not sure whether this component is really needed.

Reply:
What do you mean? You want me to move all the divs in SideNavigation.tsx?

Reply:
All one div from here 😀 What it gives to have it separately?

Reply:
oh there were two divs, I forgot I moved one already 😄 

Would just be weird to have `SidenavSubmenuHeader` but no `SidenavSubmenu`